src/driver/read.rs

```diff
@@ -11,12 +11,12 @@ pub(crate) struct Read<T> {
     #[allow(dead_code)]
     fd: SharedFd,
 
-    /// Reference to the in-flight buffer.
-    pub(crate) buf: T,
+    /// Owns the in-flight buffer.
+    buf: Slice<T>,
 }
 
 impl<T: IoBufMut> Op<Read<T>> {
-    pub(crate) fn read_at(fd: &SharedFd, buf: T, offset: u64) -> io::Result<Op<Read<T>>> {
+    pub(crate) fn read_at(fd: &SharedFd, buf: Slice<T>, offset: u64) -> io::Result<Op<Read<T>>> {
```

Comment:
I'm not sure how I feel about changing the APIs like this. Why is this necessary?

Reply:
I also feel unhappy about `Slice` popping up everywhere even when not needed, and perhaps it's not justified by fixing `.slice(..).slice(..)`.

I've got another idea which I'm going to try first.